controllers/update_pods.go

```diff
@@ -81,13 +82,35 @@ func (updatePods) reconcile(ctx context.Context, r *FoundationDBClusterReconcile
 func getPodsToUpdate(ctx context.Context, logger logr.Logger, reconciler *FoundationDBClusterReconciler, cluster *fdbv1beta2.FoundationDBCluster) (map[string][]*corev1.Pod, error) {
 	updates := make(map[string][]*corev1.Pod)
 
+	// Map of zones to the number of pods that are unavailable in that zone.
+	zoneToUnavailablePods := make(map[string]int)
+
+	// In order to the maxZonesWithUnavailablePods we need to know the number of zones.
+	// if the fault domain zone count is not set. The number of process groups is used as the zone count.
+	zoneCount := cluster.Spec.FaultDomain.ZoneCount
+	if zoneCount == 0 {
+		zoneCount = len(cluster.Status.ProcessGroups)
+	}
```

Comment:
This calculation is not correct. As I mentioned it makes sense to use an integer here instead of trying to calculate the percentage.